lib/tests.py

```diff
@@ -74,16 +80,23 @@ def test_route_for(plugin):
 
 
 def test_route_for_args(plugin):
-    f = lambda: None
-    plugin.route("/foo/<a>/<b>")(f)
-    assert plugin.route_for(plugin.base_url + "/foo/1/2") is f
+    f = lambda a, var_with_num_underscore2: None
+    g = lambda: (None, None)  # just to make sure that they are easily different
+    plugin.route("/foo/<a>/<var_with_num_underscore2>")(f)
+    plugin.route("/foo/a/b")(g)
+
+    # due to the unpredictable sorting of dict, just do it 100 times to see if it fails
```

Comment:
I doubt this is needed, dict-ordering is not guaranteed (in Python 3.5 or older) but it is deterministic. It is not random.

Reply:
I don't know about python 3, but this was actually causing issues before this PR's change, therefore I put the test to prove that it's working now

Reply:
Maybe not in Py3, but it was an issue before this PR, so it's there to prove that it's fixed with the PR

Reply:
No, I don't think this is a problem on Python 2 either. Again, the ordering is not guaranteed, but it is deterministic. So every single run for the 100 runs will be identical.

Reply:
You would think, right? I expected that, since every other language I've worked with is like that. I can tell you that, in my attempts just to find this issue, I proved the opposite on Python 2.7. When it says that it can't guarantee order there, for the first time I've seen, it really means it.

Just for the sake of it, I'll write up a test case and run it. If it yields consistent results, then I'll concede.

Reply:
```
PS C:\Users\da3ds> & "C:/Program Files/Python37/python.exe" c:/Users/da3ds/Documents/loop_test.py
apples
PS C:\Users\da3ds> & "C:/Program Files/Python37/python.exe" c:/Users/da3ds/Documents/loop_test.py
apples
PS C:\Users\da3ds> & "C:/Program Files/Python37/python.exe" c:/Users/da3ds/Documents/loop_test.py
apples
PS C:\Users\da3ds> & "C:/Program Files/Python27/python.exe" c:/Users/da3ds/Documents/loop_test.py
beetroot
PS C:\Users\da3ds> & "C:/Program Files/Python27/python.exe" c:/Users/da3ds/Documents/loop_test.py
beetroot
PS C:\Users\da3ds> & "C:/Program Files/Python27/python.exe" c:/Users/da3ds/Documents/loop_test.py
beetroot
```
You win. I don't know what was different before that was giving unreliable results, but it definitely was. That was a long time ago, a different installation of Windows, etc.